src/web/services/navigation.service.ts

```diff
@@ -24,6 +24,17 @@ export class NavigationService {
       .join('&')}`;
   }
 
+  /**
+   * Removes params keys whose values are the empty string
+   */
+  filterEmptyParams(params: {[key: string]: string}): void {
```

Comment:
What if we want to pass an argument with an empty string?

I don't think we have the case but we should not hide the removing logic in `src/web/services/navigation.service.ts`. I think if you don't want the empty string, it should not be passed into the service in the first place.

The empty string and `undefined` value is too different case.

Is there any engineering difficulties that make you do this?

Reply:
Hey Xiao Pu, this was created in order to replicate the current behaviour in `AppUrl.java`, which removes empty params when generating the links.  I thought that I should change it to match the current format. Let me know if I can change the format. 

This function is also not called on `encodeParams` by default, so the case of passing empty string is still allowed

I could move this logic back to `link.service` to make it clear that it's a link issue. What do you think?

Reply:
I think link.service is better.

Reply:
Cool, fixed!